iothub/service/src/IotHubTokenCredential.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.Amqp;
+using System.Threading;
+
+#if !NET451
+
+using Azure.Core;
+
+#endif
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// The properties required for authentication to IoT hub using a token credential.
+    /// </summary>
+    internal class IotHubTokenCredential : IotHubCredential
+    {
+#if !NET451
+
+        private const string _tokenType = "jwt";
+        private TokenCredential _credential;
```

Comment:
readonly?

Reply:
It can be for now. But we might need to add caching eventually and update it whenever it is about to expire.